astropy/wcs/utils.py

```diff
@@ -118,6 +140,60 @@ def _wcs_to_celestial_frame_builtin(wcs):
                 representation_type=SphericalRepresentation,
                 obstime=wcs.wcs.dateobs or None,
             )
+        elif (
+            "LN" in xcoord
```

Comment:
Could this be more strict, i.e., `xcoord[2:4] in ("LN", "LT") and "H" not in xcoord and "CR" not in xcoord`?